src/modding/ModManager.cs

```diff
@@ -784,18 +1170,40 @@ private void ApplyChanges()
             return;
         }
 
+        var checkResult = ModLoader.IsValidModList(enabledMods);
+        if (checkResult.ErrorType >= 0)
+        {
+            LoadEnabledMods();
+        }
+        else
+        {
+            var warningText = new LocalizedString("MOD_LOAD_ERROR_WARNING",
+                ModHelpers.CheckResultToString(checkResult, enabledMods));
+
+            loadWarningDialog.DialogText = warningText.ToString();
+            loadWarningDialog.PopupCenteredShrink();
+        }
+    }
+
+    private void LoadEnabledMods()
+    {
+        if (notEnabledMods == null || enabledMods == null)
+        {
+            return;
+        }
+
         GD.Print("Applying changes to enabled mods");
 
-        Settings.Instance.EnabledMods.Value = enabledMods.Select(m => m.InternalName).ToList();
+        Settings.Instance.EnabledMods.Value = enabledMods?.Select(m => m.InternalName).ToList() ?? new List<string>();
 
         var modLoader = ModLoader.Instance;
         modLoader.LoadMods();
 
-        var errors = modLoader.GetAndClearModErrors();
+        var errors = modLoader.GetModErrors();
 
         if (errors.Count > 0)
         {
-            modErrorDialog.ExceptionInfo = string.Join("\n", errors);
+            modErrorDialog.ExceptionInfo = string.Join("\n", errors.Select(e => e.ErrorMessage));
```

Comment:
If the errors are not cleared here, then loading multiple mods with errors, won't the errors just accumulate to the later mods. That's why I've used GetAndClearModErrors here, and also in the main menu, so that the errors don't hang around too long.